### PR TITLE
Fix determining current Kyma version by operations

### PR DESCRIPTION
**Pull request by ebensom**
**Description**

The previous [PR about exposing current Kyma version on /runtimes endpoint](https://github.com/kyma-project/control-plane/pull/1190) has multiple flaws, e.g. it doesn't take into account whether upgrade operations has been processed or not.
This PR fixes these issues.

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/internal/runtime/handler.go [hunk 1]
```diff
@@ -313,6 +305,34 @@ func (h *Handler) setRuntimeOptionalAttributes(instance internal.Instance, dto *
...
+	if len(pOprs) != 0 {
+		kymaVersion = pOprs[0].RuntimeVersion.Version
+		kymaVersionSetAt = pOprs[0].CreatedAt
+	}
+
+	// Take the last upgrade kyma operation which
+	//   - is not dry-run
+	//   - is created after the last provisioning operation
+	//   - has the kyma version set
+	//   - the has been processed, i.e. not pending, canceling or canceled
```
Comment (wozniakjan) [inline components/kyma-environment-broker/internal/runtime/handler.go:322]: nit: I would drop "the" from "the has been..."

**Review by wozniakjan — DISMISSED** [review]
/lgtm
/hold

feel free to unhold if you are ok with the nit below or ping me for reapproval

File: components/kyma-environment-broker/internal/runtime/handler.go (hunk 1 above)
Comment (ebensom) [reply components/kyma-environment-broker/internal/runtime/handler.go]: Fixed.

**Review by wozniakjan — APPROVED** [review]
/lgtm

**Pull request merged**